opensrp-chw/src/androidTest/java/org/smartregister/chw/activity/ba/ANCRegisterTestsBa.java

```diff
@@ -0,0 +1,144 @@
+    @Rule
+    public ActivityTestRule<LoginActivity> mActivityTestRule = new ActivityTestRule<>(LoginActivity.class);
+
+    @Rule
+    public GrantPermissionRule permissionRule = GrantPermissionRule.grant(android.Manifest.permission.CALL_PHONE);
+
+    @Rule
+    public GrantPermissionRule mRuntimePermissionRule = GrantPermissionRule.grant(Manifest.permission.CAMERA);
+
+    @Rule
+    public GrantPermissionRule mRuntimePermissionRule1 = GrantPermissionRule.grant(Manifest.permission.ACCESS_FINE_LOCATION);
+
+    private Utils utils = new Utils();
+
+    public void setUp() throws InterruptedException {
+        utils.logIn(Constants.BoreshaAfyaConfigUtils.ba_username, Constants.BoreshaAfyaConfigUtils.ba_password);
+    }
+
+    @Test
+    @Order(order = 1)
+    public void searchANCRecord() {
+        utils.openDrawer();
+        onView(ViewMatchers.withSubstring(Constants.GenericConfigUtils.anc))
+                .perform(click());
+        onView(withHint("Search name or ID"))
+                .perform(typeText(Configs.TestConfigHelper.aboveFiveFirstNameTwo), closeSoftKeyboard());
+        onView(ViewMatchers.withSubstring(Configs.TestConfigHelper.aboveFiveFirstNameTwo
+                + " " + Configs.TestConfigHelper.aboveFiveSecondNameTwo))
+                .check(matches(isDisplayed()));
+    }
+    @Test
+    @Order(order = 2)
+    public void confirmANCRecordProfile() {
+        utils.openDrawer();
+        onView(ViewMatchers.withSubstring(Constants.GenericConfigUtils.anc))
+                .perform(click());
+        onView(withHint("Search name or ID"))
+                .perform(typeText(Configs.TestConfigHelper.aboveFiveFirstNameTwo), closeSoftKeyboard());
+        onView(ViewMatchers.withSubstring(Configs.TestConfigHelper.aboveFiveFirstNameTwo
+                + " " + Configs.TestConfigHelper.aboveFiveSecondNameTwo))
+                .check(matches(isDisplayed()));
+    }
+
+    @Test
+    @Order(order = 3)
+    public void confirmANCVisitPage() {
+        utils.openDrawer();
+        onView(ViewMatchers.withSubstring(Constants.GenericConfigUtils.anc))
+                .perform(click());
+        onView(withHint("Search name or ID"))
+                .perform(typeText(Configs.TestConfigHelper.aboveFiveFirstNameTwo), closeSoftKeyboard());
+        onView(ViewMatchers.withSubstring(Configs.TestConfigHelper.aboveFiveFirstNameTwo
+                + " " + Configs.TestConfigHelper.aboveFiveSecondNameTwo))
+                .perform(click());
+        onView(withId(R.id.textview_record_visit))
+                .perform(click());
+        onView(ViewMatchers.withSubstring(Configs.TestConfigHelper.aboveFiveFirstNameTwo
+                + " " + Configs.TestConfigHelper.aboveFiveSecondNameTwo + " " + Configs.TestConfigHelper.familyName + ", "+ Configs.TestConfigHelper.aboveFiveage + " · ANC Visit"))
+                .check(matches(isDisplayed()));
+    }
+    @Test
+    @Order(order = 4)
+    public void confirmANCSearchWorks() {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [JUnit tests should include assert() or fail()](https://app.codacy.com/manual/OpenSRP/opensrp-unicef-wcaro/pullRequest?prid=5426621)